### ISSUE TITLE
Changing epochs

### ISSUE THREAD

**Issue by golnaz-hs** [MIC-DKFZ/nnUNet#1309]
Hello,
I want to change the number of epochs in TrainerV2.  So I set the new number of epochs in the custom class self.max_num_epochs = 100. but TrainerV2 still runs in 1000 epochs. 

Many thanks for your time ...

**Comment by FabianIsensee** [comment]
Can you be more specific please? 
This should be really simple:

'''
class nnUNetTrainerV2_2epochs(nnUNetTrainerV2):
    def __init__(self, plans_file, fold, output_folder=None, dataset_directory=None, batch_dice=True, stage=None,
                 unpack_data=True, deterministic=True, fp16=False):
        super().__init__(plans_file, fold, output_folder, dataset_directory, batch_dice, stage, unpack_data,
                         deterministic, fp16)
        self.max_num_epochs = 2
'''

**Comment by FabianIsensee** [comment]
Hey, this is not how you do inheritance in python. Take a look here: https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py

**Comment by darshitabudhadev** [comment]
Hi @FabianIsensee,  Thanks for nnunetv2. 
I am also trying to reduce the number of epochs as after 48 h of running the nnunetv2 model on my custom tomogram dataset in GPU(cudnn_version: 8500), it only completes 216 epochs on 3d_fullres configuration. I only get 48 hr of GPU time. So as per your suggestion, I added an extra class in: "/nnUNet/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py" for 200 epochs as:
class nnUNetTrainer_200epochs(nnUNetTrainer):
    def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, unpack_dataset: bool = True,
                 device: torch.device = torch.device('cuda')):
        super().__init__(plans, configuration, fold, dataset_json, unpack_dataset, device)
        self.num_epochs = 200
Then for training I used: CUDA_VISIBLE_DEVICES=0,1 nnUNetv2_train $DATASET_NAME_OR_ID $DATATYPE3D $FOLD -tr nnUNetTrainer_200epochs  -device 'cuda' -num_gpus 2
But it gives error: Could not find requested nnunet trainer nnUNetTrainer_200epochs in nnunetv2.training.nnUNetTrainer
Am I doing something wrong here? Any help would be really appreciated. 
Thanks in advance..
